ios/browser/brave_ads/virtual_pref_provider_delegate_ios.mm

```diff
@@ -0,0 +1,38 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/ios/browser/brave_ads/virtual_pref_provider_delegate_ios.h"
+
+#include "base/strings/utf_string_conversions.h"
+#include "base/version_info/channel.h"
+#include "components/search_engines/template_url_data.h"
+#include "components/search_engines/template_url_prepopulate_data.h"
+#include "components/search_engines/template_url_prepopulate_data_resolver.h"
+#include "ios/chrome/browser/search_engines/model/template_url_prepopulate_data_resolver_factory.h"
+#include "ios/chrome/browser/shared/model/profile/profile_ios.h"
+#include "ios/chrome/common/channel_info.h"
+
+namespace brave_ads {
+
+VirtualPrefProviderDelegateIOS::VirtualPrefProviderDelegateIOS(
+    ProfileIOS& profile)
+    : profile_(profile) {}
+
+VirtualPrefProviderDelegateIOS::~VirtualPrefProviderDelegateIOS() = default;
+
+std::string VirtualPrefProviderDelegateIOS::GetChannel() const {
+  return std::string(version_info::GetChannelString(::GetChannel()));
+}
+
+std::string VirtualPrefProviderDelegateIOS::GetDefaultSearchEngineName() const {
+  auto* prepopulate_data_resolver =
```

Comment:
Does Brave iOS use the chromium code to select the default engine?

Reply:
@tmancey No

Reply:
Gotcha, so this part should be re-written then. I'm converting the current PR to Draft until this part is finished.

Reply:
Currently `VirtualPrefProviderDelegateIOS::GetDefaultSearchEngineName` function uses Brave chromium override to get default search engine. The list of default search engines per regions is here: https://github.com/brave/brave-core/blob/master/chromium_src/components/regional_capabilities/regional_capabilities_utils.cc#L641

And it is the same as default search engines list for iOS from here: https://github.com/brave/brave-core/blob/master/ios/brave-ios/Sources/Brave/Frontend/Browser/Search/InitialSearchEngines.swift#L98.

So basing on discussion in DM with @kylehickinson and it was decided that we can go with the current implementation because these lists will always be same and eventually iOS will move to chromium implementation.